src/dune_rules/odoc.ml

```diff
@@ -243,6 +259,51 @@ module Flags = struct
   ;;
 end
 
+module Version = struct
+  type t = int * int * int (* major * minor * patch *)
+
+  let of_string s : t option =
```

Comment:
We have the `re` library available that might make this version parsing simpler.